benchmark/vochain_benchmark_test.go

```diff
@@ -273,19 +194,66 @@ func voteBench(b *testing.B, cl *client.Client, s *ethereum.SignKeys,
 		b.Fatal(err)
 	}
 	// sending submitEnvelope request
-	resp = doRequest("submitRawTx", nil)
-	log.Infof("response: %s", resp.String())
+	resp := doRequest("submitRawTx", nil)
+	if !resp.Ok {
+		b.Errorf("error submitting vote for %s: %s", s.AddressString(), resp.String())
+	}
+}
 
-	// check vote added
-	/*	reset(req)
-		req.ProcessID = processID
-		req.Nullifier = vochain.GenerateNullifier(s.Address(), processID)
-		for {
-			resp = doRequest("getEnvelopeStatus", nil)
-			if *resp.Registered {
-				break
+func waitProcess(b *testing.B, cl *client.Client, processID []byte) {
+	log.Info("waiting for the process to start")
+	req := &api.APIrequest{}
+	reset(req)
+	doRequest := cl.ForTest(b, req)
+	req.ProcessID = processID
+	failures := 20
+	for {
+		procInfo := doRequest("getProcessInfo", nil)
+		if procInfo.Ok {
+			if procInfo.Process.Status == int32(models.ProcessStatus_READY) {
+				height := doRequest("getBlockHeight", nil)
+				if *height.Height >= procInfo.Process.StartBlock {
+					break
+				}
 			}
-			time.Sleep(time.Second * 2)
 		}
-	*/
+		failures--
+		if failures == 0 {
+			b.Fatalf("processID %s should be started by now", hexProcessID)
```

Comment:
This Fatal can be substituted by an error and we can execute the fatal in an upper layer. So `waitProcess(...args) error`